lib/post_voting/comment_review_queue.rb

```diff
@@ -0,0 +1,151 @@
+# frozen_string_literal: true
+module PostVoting
+  class CommentReviewQueue
+    def flag_comment(comment, guardian, flag_type_id, opts = {})
+      result = { success: false, errors: [] }
+      is_notify_user_type = ReviewableScore.types.slice(:notify_user).values.include?(flag_type_id)
+      is_notify_moderators_type =
+        ReviewableScore.types.slice(:notify_moderators).values.include?(flag_type_id)
+      is_notify_type = is_notify_user_type || is_notify_moderators_type
+
+      guardian.ensure_can_flag_post_voting_comment!(comment)
+      guardian.ensure_can_flag_post_voting_comment_as!(comment, flag_type_id, opts)
```

Comment:
This seems odd, is there any way we could condense this logic, or maybe implement some better naming?